grpc-example/grpc-example-java/hello-proxy-impl/pom.xml

```diff
@@ -69,6 +69,7 @@
                 <artifactId>lagom-maven-plugin</artifactId>
                 <configuration>
                     <lagomService>true</lagomService>
+                    <serviceEnableSsl>true</serviceEnableSsl>
```

Comment:
This is the actual fix. In order for a client-side to trust the self-signed certificates used in dev mode, it is necessary to `enableSsl`  on the service (even if that service is not binding a TLS port (aka HTTPS).